plugins/generator-1.19.4/forge-1.19.4/templates/livingentity/livingentity.java.ftl

```diff
@@ -161,19 +171,25 @@ public class ${name}Entity extends ${extendsClass} <#if data.ranged>implements R
 	}
 	</#if>
 
-	<#if data.hasAI>
+	<#if data.hasAI || data.hasWanderingTraderTrade()>
 	@Override protected void registerGoals() {
 		super.registerGoals();
 
-		<#if aicode??>
+		<#if data.hasWanderingTraderTrade()>
+			this.goalSelector.addGoal(1, new TradeWithPlayerGoal(this));
+			this.goalSelector.addGoal(1, new LookAtTradingPlayerGoal(this));
+			this.goalSelector.addGoal(10, new LookAtPlayerGoal(this, Mob.class, 8.0F));
+		</#if>
+
+		<#if data.hasAI && aicode??>
 			<#if aiblocks?seq_contains("doors_open") || aiblocks?seq_contains("doors_close")>
 				this.getNavigation().getNodeEvaluator().setCanOpenDoors(true);
 			</#if>
 
             ${aicode}
         </#if>
 
-        <#if data.ranged>
+        <#if data.hasAI && data.ranged>

```

Comment:
This change should be own PR as it is unrelated to this one

Reply:
This makes sure the code inside the `if` is not executed if the mob has a profession.

Reply:
But in all cases makes sense to not add AI tasks if mod has no AI enabled.

If intent was 

> This makes sure the code inside the if is not executed if the mob has a profession.

this code does not communicate this intent well

Reply:
The reason I added `data.hasAI` to the condition is because the small condition is in a bigger condition that can be accessed even if the entity doesn't have an AI, but has trades (e.g. bigger condition begins line 174) so you need to double check that the entity is ranged AND does have AI.

Reply:
Ok I removed the condition then

Reply:
I did not mean to remove the condition, but to open another PR that adds this condition so it is added to the code before this PR is merged because

> I still think we would benefit from this change outside this PR too

Now that you removed it you removed it while:

* it is still needed in your PR
* I also said we need to outside PR

Removal does not address any of those

Reply:
What I don't get is why this extra condition is needed without my changes. Did you see the raw code, outside of my PR? The `if data.ranged` condition is not accessed before the `if data.hasAI` condition, which is not the case with my changes.

Reply:
I see